app/src/org/commcare/dalvik/activities/UnrecoverableErrorActivity.java

```diff
@@ -43,7 +39,7 @@ protected Dialog onCreateDialog(int id) {
         AlertDialogFactory factory = new AlertDialogFactory(this, title, message);
         DialogInterface.OnClickListener buttonListener = new DialogInterface.OnClickListener() {
             public void onClick(DialogInterface dialog, int i) {
-               Intent intent = new Intent(UnrecoverableErrorActivity.this, CommCareHomeActivity.class);
+               Intent intent = new Intent(UnrecoverableErrorActivity.this, DispatchActivity.class);

```

Comment:
Most of the irrecoverable errors here are unrecoverable in the sense that there's no clear/safe action to follow to get the user back into a state where we are confident that things aren't going to commit bad state. It doesn't necessarily mean "The app can't run", just that we can't be sure that, say, logging you out or something similar is enough to put things right.